_modules/vault.py

```diff
@@ -177,16 +177,32 @@ def cached_read(path, cache_prefix='', **kwargs):
             lease_valid = True
         else:
             lease_valid = False
-            vault_client.delete(cache_path)
-
-    if not vault_data or not lease_valid:
+            if not vault_data['renewable']:
```

Comment:
I would add an extra condition based on a keyword arg so we can override this behavior in the event that we don't want to renew leases for certain situations.

Reply:
In looking at how you're handling `lease_valid` I'm wondering about the case when `if not vault_data or not lease_valid` [is met](https://github.com/mitodl/vault-formula/blob/master/_modules/vault.py#L182). Shouldn't that throw an undefined error for the `lease_valid` keyword since that gets defined only when `vault_data` is non-empty?

Reply:
The `or` operator is a short-circuit logic path so it won't every try to evaluate `lease_valid` in the event that there is no vault data.